src/system_data.jl

```diff
@@ -311,6 +311,27 @@ function clear_time_series!(data::SystemData)
     reset_info!(data.time_series_params)
 end
 
+"""
+Removes all time series of a particular type from a System.
+
+# Arguments
+- `data::SystemData`: system
+- `type::Type{<:TimeSeriesData}`: Type of time series objects to remove.
+"""
+function remove_time_series!(data::SystemData, ::Type{T}) where {T <: TimeSeriesData}
+    for component in iterate_components_with_time_series(data.components)
+        for ts in get_time_series_multiple(component, type = T)
+            remove_time_series!(data, typeof(ts), component, get_name(ts))
+        end
+    end
+    counts = get_time_series_counts(data)
+    if counts[2] == 0 # no more time series objects
+        reset_info!(data.time_series_params)
```

Comment:
This will reset static and forecasts. The main struct defines resolution which is common to both.